// @strapi/upload/dist/admin/future/pages/Assets/AssetsPage.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var React = require('react');
var ToggleGroup = require('@radix-ui/react-toggle-group');
var strapiAdmin = require('@strapi/admin/strapi-admin');
var designSystem = require('@strapi/design-system');
var icons = require('@strapi/icons');
var reactIntl = require('react-intl');
var styledComponents = require('styled-components');
var usePersistentState = require('../../../hooks/usePersistentState.js');
var api = require('../../services/api.js');
var assets = require('../../services/assets.js');
var translations = require('../../utils/translations.js');
var AssetsGrid = require('./components/AssetsGrid.js');
var AssetsTable = require('./components/AssetsTable.js');
var DropZoneWithOverlay = require('./components/DropZone/DropZoneWithOverlay.js');
var UploadDropZoneContext = require('./components/DropZone/UploadDropZoneContext.js');
var constants = require('./constants.js');

function _interopNamespaceDefault(e) {
  var n = Object.create(null);
  if (e) {
    Object.keys(e).forEach(function (k) {
      if (k !== 'default') {
        var d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: function () { return e[k]; }
        });
      }
    });
  }
  n.default = e;
  return Object.freeze(n);
}

var ToggleGroup__namespace = /*#__PURE__*/_interopNamespaceDefault(ToggleGroup);

const AssetsView = ({ view })=>{
    const { formatMessage } = reactIntl.useIntl();
    const { data, isLoading, error } = assets.useGetAssetsQuery({
        folder: null
    });
    const isGridView = view === constants.viewOptions.GRID;
    const assets$1 = data?.results ?? [];
    if (isLoading) {
        return /*#__PURE__*/ jsxRuntime.jsx(designSystem.Flex, {
            justifyContent: "center",
            padding: 8,
            children: /*#__PURE__*/ jsxRuntime.jsx(designSystem.Loader, {
                children: formatMessage({
                    id: 'app.loading',
                    defaultMessage: 'Loading...'
                })
            })
        });
    }
    if (error) {
        return /*#__PURE__*/ jsxRuntime.jsx(designSystem.Box, {
            padding: 8,
            children: /*#__PURE__*/ jsxRuntime.jsx(designSystem.Typography, {
                textColor: "danger600",
                children: formatMessage({
                    id: translations.getTranslationKey('list.assets.error'),
                    defaultMessage: 'An error occurred while fetching assets.'
                })
            })
        });
    }
    if (isGridView) {
        return /*#__PURE__*/ jsxRuntime.jsx(AssetsGrid.AssetsGrid, {
            assets: assets$1
        });
    }
    return /*#__PURE__*/ jsxRuntime.jsx(AssetsTable.AssetsTable, {
        assets: assets$1
    });
};
const StyledToggleGroup = styledComponents.styled(ToggleGroup__namespace.Root)`
  display: flex;
  border: 1px solid ${({ theme })=>theme.colors.neutral200};
  border-radius: ${({ theme })=>theme.borderRadius};
  overflow: hidden;
`;
const StyledToggleItem = styledComponents.styled(ToggleGroup__namespace.Item)`
  display: flex;
  align-items: center;
  gap: ${({ theme })=>theme.spaces[2]};
  padding: ${({ theme })=>`${theme.spaces[2]} ${theme.spaces[4]}`};
  border: none;
  background: ${({ theme })=>theme.colors.neutral0};
  color: ${({ theme })=>theme.colors.neutral800};
  cursor: pointer;
  font-size: ${({ theme })=>theme.fontSizes[1]};
  font-weight: ${({ theme })=>theme.fontWeights.semiBold};

  &:hover {
    background: ${({ theme })=>theme.colors.neutral100};
  }

  &[data-state='on'] {
    background: ${({ theme })=>theme.colors.neutral150};
  }

  svg {
    width: 1.6rem;
    height: 1.6rem;
  }
`;
const DropFilesMessage = ({ uploadDropZoneRef })=>{
    const { formatMessage } = reactIntl.useIntl();
    const { isDragging } = UploadDropZoneContext.useUploadDropZone();
    // Dropzone message position (relative to main content)
    const [leftContentWidth, setLeftContentWidth] = React.useState(0);
    // Calculate the left content width to position the dropzone message correctly
    React.useEffect(()=>{
        if (!uploadDropZoneRef?.current) return;
        const updateRect = ()=>{
            const rect = uploadDropZoneRef.current?.getBoundingClientRect();
            if (rect) {
                setLeftContentWidth((prev)=>prev !== rect.left ? rect.left : prev);
            }
        };
        updateRect();
        const resizeObserver = new ResizeObserver(updateRect);
        resizeObserver.observe(uploadDropZoneRef.current);
        return ()=>resizeObserver.disconnect();
    }, [
        uploadDropZoneRef
    ]);
    if (!isDragging) return null;
    return /*#__PURE__*/ jsxRuntime.jsxs(DropFilesMessageImpl, {
        $leftContentWidth: leftContentWidth,
        children: [
            /*#__PURE__*/ jsxRuntime.jsx(designSystem.Typography, {
                textColor: "neutral0",
                children: formatMessage({
                    id: translations.getTranslationKey('dropzone.upload.message'),
                    defaultMessage: 'Drop here to upload to'
                })
            }),
            /*#__PURE__*/ jsxRuntime.jsxs(designSystem.Flex, {
                gap: 2,
                alignItems: "center",
                children: [
                    /*#__PURE__*/ jsxRuntime.jsx(icons.Folder, {
                        width: 20,
                        height: 20,
                        fill: "neutral0"
                    }),
                    /*#__PURE__*/ jsxRuntime.jsx(designSystem.Typography, {
                        textColor: "neutral0",
                        fontWeight: "semiBold",
                        children: "Current folder"
                    })
                ]
            })
        ]
    });
};
const DropFilesMessageImpl = styledComponents.styled(designSystem.Box)`
  position: fixed;
  bottom: ${({ theme })=>theme.spaces[8]};
  left: 50%;
  transform: translateX(calc(-50% + ${({ $leftContentWidth })=>$leftContentWidth / 2}px));
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: ${({ theme })=>theme.spaces[2]};
  background: ${({ theme })=>theme.colors.primary600};
  padding: ${({ theme })=>theme.spaces[4]} ${({ theme })=>theme.spaces[6]};
  border-radius: ${({ theme })=>theme.borderRadius};
  z-index: 2;
`;
const AssetsPage = ()=>{
    const { formatMessage } = reactIntl.useIntl();
    // View state
    const [view, setView] = usePersistentState.usePersistentState(constants.localStorageKeys.view, constants.viewOptions.GRID);
    const isGridView = view === constants.viewOptions.GRID;
    // Refs
    const fileInputRef = React.useRef(null);
    const uploadDropZoneRef = React.useRef(null);
    // Upload handlers
    const { toggleNotification } = strapiAdmin.useNotification();
    const { _unstableFormatAPIError } = strapiAdmin.useAPIErrorHandler();
    const [uploadFiles] = api.useUploadFilesMutation();
    const uploadFilesToFolder = async (files, folderId)=>{
        if (files.length === 0) return;
        const formData = new FormData();
        files.forEach((file)=>{
            formData.append('files', file);
            formData.append('fileInfo', JSON.stringify({
                name: file.name,
                caption: null,
                alternativeText: null,
                folder: folderId
            }));
        });
        try {
            await uploadFiles(formData).unwrap();
            toggleNotification({
                type: 'success',
                message: formatMessage({
                    id: translations.getTranslationKey('assets.uploaded'),
                    defaultMessage: '{number, plural, one {# asset} other {# assets}} uploaded successfully'
                }, {
                    number: files.length
                })
            });
        } catch (error) {
            const errorMessage = _unstableFormatAPIError(error);
            toggleNotification({
                type: 'danger',
                message: errorMessage
            });
        }
    };
    const handleFileSelect = ()=>{
        fileInputRef.current?.click();
    };
    const handleFileChange = async (e)=>{
        const files = e.target.files;
        if (files && files.length > 0) {
            await uploadFilesToFolder(Array.from(files), null);
        }
        e.target.value = '';
    };
    const handleDrop = async (files)=>{
        await uploadFilesToFolder(files, null);
    };
    return /*#__PURE__*/ jsxRuntime.jsx(UploadDropZoneContext.UploadDropZoneProvider, {
        onDrop: handleDrop,
        children: /*#__PURE__*/ jsxRuntime.jsx(designSystem.Box, {
            ref: uploadDropZoneRef,
            children: /*#__PURE__*/ jsxRuntime.jsxs(strapiAdmin.Layouts.Root, {
                minHeight: "100vh",
                children: [
                    /*#__PURE__*/ jsxRuntime.jsx(designSystem.VisuallyHidden, {
                        children: /*#__PURE__*/ jsxRuntime.jsx("input", {
                            type: "file",
                            ref: fileInputRef,
                            onChange: handleFileChange,
                            multiple: true
                        })
                    }),
                    /*#__PURE__*/ jsxRuntime.jsx(DropFilesMessage, {
                        uploadDropZoneRef: uploadDropZoneRef
                    }),
                    /*#__PURE__*/ jsxRuntime.jsx(strapiAdmin.Layouts.Header, {
                        navigationAction: /*#__PURE__*/ jsxRuntime.jsx(designSystem.Box, {
                            children: "TODO: Breadcrumbs"
                        }),
                        title: "TODO: Folder location",
                        primaryAction: /*#__PURE__*/ jsxRuntime.jsxs(designSystem.Flex, {
                            gap: 2,
                            children: [
                                /*#__PURE__*/ jsxRuntime.jsx(designSystem.SimpleMenu, {
                                    popoverPlacement: "bottom-end",
                                    variant: "default",
                                    endIcon: /*#__PURE__*/ jsxRuntime.jsx(icons.ChevronDown, {}),
                                    label: formatMessage({
                                        id: translations.getTranslationKey('new'),
                                        defaultMessage: 'New'
                                    }),
                                    children: /*#__PURE__*/ jsxRuntime.jsx(designSystem.MenuItem, {
                                        onSelect: handleFileSelect,
                                        startIcon: /*#__PURE__*/ jsxRuntime.jsx(icons.Files, {}),
                                        children: formatMessage({
                                            id: translations.getTranslationKey('import-files'),
                                            defaultMessage: 'Import files'
                                        })
                                    })
                                }),
                                /*#__PURE__*/ jsxRuntime.jsx(strapiAdmin.SearchInput, {
                                    label: formatMessage({
                                        id: translations.getTranslationKey('search.label'),
                                        defaultMessage: 'Search for an asset'
                                    }),
                                    trackedEvent: "didSearchMediaLibraryElements",
                                    trackedEventDetails: {
                                        location: 'upload'
                                    }
                                }),
                                /*#__PURE__*/ jsxRuntime.jsxs(StyledToggleGroup, {
                                    type: "single",
                                    value: isGridView ? 'grid' : 'table',
                                    onValueChange: (value)=>value && setView(value === 'grid' ? constants.viewOptions.GRID : constants.viewOptions.TABLE),
                                    "aria-label": formatMessage({
                                        id: translations.getTranslationKey('view.switch.label'),
                                        defaultMessage: 'View options'
                                    }),
                                    children: [
                                        /*#__PURE__*/ jsxRuntime.jsxs(StyledToggleItem, {
                                            value: "table",
                                            "aria-label": formatMessage({
                                                id: translations.getTranslationKey('view.table'),
                                                defaultMessage: 'Table view'
                                            }),
                                            children: [
                                                /*#__PURE__*/ jsxRuntime.jsx(icons.List, {}),
                                                formatMessage({
                                                    id: translations.getTranslationKey('view.table'),
                                                    defaultMessage: 'Table view'
                                                })
                                            ]
                                        }),
                                        /*#__PURE__*/ jsxRuntime.jsxs(StyledToggleItem, {
                                            value: "grid",
                                            "aria-label": formatMessage({
                                                id: translations.getTranslationKey('view.grid'),
                                                defaultMessage: 'Grid view'
                                            }),
                                            children: [
                                                /*#__PURE__*/ jsxRuntime.jsx(icons.GridFour, {}),
                                                formatMessage({
                                                    id: translations.getTranslationKey('view.grid'),
                                                    defaultMessage: 'Grid view'
                                                })
                                            ]
                                        })
                                    ]
                                })
                            ]
                        })
                    }),
                    /*#__PURE__*/ jsxRuntime.jsx(strapiAdmin.Layouts.Content, {
                        children: /*#__PURE__*/ jsxRuntime.jsx(DropZoneWithOverlay.DropZoneWithOverlay, {
                            children: /*#__PURE__*/ jsxRuntime.jsx(AssetsView, {
                                view: view
                            })
                        })
                    })
                ]
            })
        })
    });
};

exports.AssetsPage = AssetsPage;
