// @strapi/upload/dist/admin/future/App.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
require('react');
var strapiAdmin = require('@strapi/admin/strapi-admin');
var reactIntl = require('react-intl');
var reactRouterDom = require('react-router-dom');
var AIGenerationPage = require('./pages/AIGenerationPage.js');
var AssetsPage = require('./pages/Assets/AssetsPage.js');
var translations = require('./utils/translations.js');

const UnstableMediaLibrary = ()=>{
    const { formatMessage } = reactIntl.useIntl();
    const title = formatMessage({
        id: translations.getTranslationKey('plugin.name'),
        defaultMessage: 'Media Library'
    });
    return /*#__PURE__*/ jsxRuntime.jsxs(strapiAdmin.Page.Main, {
        children: [
            /*#__PURE__*/ jsxRuntime.jsx(strapiAdmin.Page.Title, {
                children: title
            }),
            /*#__PURE__*/ jsxRuntime.jsxs(reactRouterDom.Routes, {
                children: [
                    /*#__PURE__*/ jsxRuntime.jsx(reactRouterDom.Route, {
                        index: true,
                        element: /*#__PURE__*/ jsxRuntime.jsx(AssetsPage.AssetsPage, {})
                    }),
                    /*#__PURE__*/ jsxRuntime.jsx(reactRouterDom.Route, {
                        path: "ai-generation",
                        element: /*#__PURE__*/ jsxRuntime.jsx(AIGenerationPage.AIGenerationPage, {})
                    })
                ]
            })
        ]
    });
};

exports.UnstableMediaLibrary = UnstableMediaLibrary;
