// @strapi/upload/dist/admin/future/App.mjs
import { jsxs, jsx } from 'react/jsx-runtime';
import 'react';
import { Page } from '@strapi/admin/strapi-admin';
import { useIntl } from 'react-intl';
import { Routes, Route } from 'react-router-dom';
import { AIGenerationPage } from './pages/AIGenerationPage.mjs';
import { AssetsPage } from './pages/Assets/AssetsPage.mjs';
import { getTranslationKey } from './utils/translations.mjs';

const UnstableMediaLibrary = ()=>{
    const { formatMessage } = useIntl();
    const title = formatMessage({
        id: getTranslationKey('plugin.name'),
        defaultMessage: 'Media Library'
    });
    return /*#__PURE__*/ jsxs(Page.Main, {
        children: [
            /*#__PURE__*/ jsx(Page.Title, {
                children: title
            }),
            /*#__PURE__*/ jsxs(Routes, {
                children: [
                    /*#__PURE__*/ jsx(Route, {
                        index: true,
                        element: /*#__PURE__*/ jsx(AssetsPage, {})
                    }),
                    /*#__PURE__*/ jsx(Route, {
                        path: "ai-generation",
                        element: /*#__PURE__*/ jsx(AIGenerationPage, {})
                    })
                ]
            })
        ]
    });
};

export { UnstableMediaLibrary };
