// @strapi/upload/dist/admin/future/pages/Assets/components/DropZone/DropZoneWithOverlay.js
'use strict';

var jsxRuntime = require('react/jsx-runtime');
var designSystem = require('@strapi/design-system');
var styledComponents = require('styled-components');
var UploadDropZoneContext = require('./UploadDropZoneContext.js');

const setOpacity = (hex, alpha)=>`${hex}${Math.floor(alpha * 255).toString(16).padStart(2, '0')}`;
const DropZoneOverlay = styledComponents.styled(designSystem.Box)`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: ${({ theme })=>setOpacity(theme.colors.primary200, 0.3)};
  border: 1px solid ${({ theme })=>theme.colors.primary700};
  border-radius: ${({ theme })=>theme.borderRadius};
  z-index: 1;
  pointer-events: none;
`;
const DropZoneWithOverlay = ({ children })=>{
    const { isDragging } = UploadDropZoneContext.useUploadDropZone();
    return /*#__PURE__*/ jsxRuntime.jsxs(designSystem.Box, {
        position: "relative",
        children: [
            isDragging && /*#__PURE__*/ jsxRuntime.jsx(DropZoneOverlay, {}),
            children
        ]
    });
};

exports.DropZoneWithOverlay = DropZoneWithOverlay;
